assets/js/nav-toggle.js

```diff
@@ -62,6 +62,12 @@ window.addEventListener(
 window.addEventListener(
     'resize',
     throttle(function() {
+        // https://github.com/reuixiy/hugo-theme-meme/issues/171
+        const input = document.getElementById('search-input');
+        if (input && input === document.activeElement) {
+            return;
+        }
+
         closeNav();
```

Comment:
Why are we doing this in the first place? I guess the idea is to get rid of artifacts when the user makes the window bigger and the nav toggle is no longer visible? Then maybe we should check `document.querySelector(".nav-toggle").offsetWidth` and call `closeNav()` if it's zero, completely independent of the search fields.

Reply:
> Why are we doing this in the first place? I guess the idea is to get rid of artifacts when the user makes the window bigger and the nav toggle is no longer visible?

The `resize` listener? Yes.

> Then maybe we should check `document.querySelector(".nav-toggle").offsetWidth` and call `closeNav()` if it's zero, completely independent of the search fields.

Nope, we check this inside of the `closeNav()` function, see [here](https://github.com/reuixiy/hugo-theme-meme/blob/4c6097a906dac7c4109445b4f2d63574b53b155a/assets/js/nav-toggle.js#L70).

This `if`, however, prevents the drop-down menu from being collapsed by a `resize` event produced by the popped keyboard when search field is focused - although I don't know why it exists.

Reply:
> Nope, we check this inside of the `closeNav()` function, see [here](https://github.com/reuixiy/hugo-theme-meme/blob/4c6097a906dac7c4109445b4f2d63574b53b155a/assets/js/nav-toggle.js#L70).

I didn't mean checking whether the menu is open - I meant checking whether the nav toggle is still visible in general. The `.nav-toggle` element is only visible when the window size is small, we can check that and close the menu only if it isn't visible any more (window size got large enough to display the menu permanently).

Reply:
I got little puzzled... Could you create a new commit for this?

Reply:
Done, does it make sense now? Note that I had to remove the fade logic - there is no fading animation in this scenario, so the nav curtain remains visible.

Reply:
Reproduced the issue with nav curtain remaining visible when resizing a small browser window on palant.info in both Firefox and Chrome.

Reply:
Alternatively, it should be possible to listen to the `(max-width: $maxWidth)` media query instead of window resizes. One complication that probably makes it not worth doing: with this variable being calculated in SASS, it would have to be exposed somewhere - e.g. as a CSS variable on the document root.

Reply:
Nice! Works like a charm! I'll try the media query approach.